### PR TITLE
DOC-7805: Improve overview of CBO

### PR DESCRIPTION
**Pull request by simon-dew**
The following draft documentation is ready for review:

* [Cost-Based Optimizer](https://simon-dew.github.io/docs-site/DOC-7925/server/7.0/n1ql/n1ql-language-reference/cost-based-optimizer.html)
  - Better introduction.
  - Emphasise that users need to update statistics.
  - Activating and deactivating the optimizer.
  - Explain how the optimizer fits into query plans.
  - Update diagrams.
  
 Docs issue: [DOC-7805](https://issues.couchbase.com/browse/DOC-7805)

### CODE REVIEW

File: modules/n1ql/pages/n1ql-language-reference/cost-based-optimizer.adoc [hunk 1]
```diff
@@ -24,14 +24,15 @@ The cost-based optimizer takes into account the cost of memory, CPU, network tra
 [[overview]]
 == Overview
 
-The _cost-based optimizer_ (CBO) replaces the previous {query-service-architecture}[rules-based N1QL optimizer].
-It can generate a query plan for {select}[SELECT], {update}[UPDATE], {delete}[DELETE], {merge}[MERGE], and {insert}[INSERT INTO with SELECT] queries.
+The _cost-based optimizer_ (CBO) is a tool that enables the Query service to find the most efficient way of fetching data and performing a query.
```
Comment (keshavmr) [inline modules/n1ql/pages/n1ql-language-reference/cost-based-optimizer.adoc]: It's not a tool in a "product" sense... Its job is to create the most efficient plan to execute the query.  See: https://dzone.com/articles/less-is-more-why-cost-based-optimizer. and https://dzone.com/articles/cost-based-optimizer-for-couchbase-n1ql-sql-for-js

**Comment by miaobingjie** [comment]
_The cost-based optimizer does not yet rewrite the query to use the optimal join ordering or join type._

This is not accurate. CBO does not reorder joins yet, but it does choose the optimal join type for each join.

File: modules/n1ql/pages/n1ql-language-reference/cost-based-optimizer.adoc (hunk 1 above)
Comment (simon-dew) [reply modules/n1ql/pages/n1ql-language-reference/cost-based-optimizer.adoc]: Done

**Comment by simon-dew** [comment]
> CBO does not reorder joins yet, but it does choose the optimal join type for each join.

Done

**Pull request merged**